engine/kv_engine.cpp

```diff
@@ -792,8 +792,10 @@ Status KVEngine::HashGetImpl(const pmem::obj::string_view &key,
     }
 
     // Copy PMem data record to dram buffer
-    // TODO jiayu: record_size maybe invalid here?
-    auto record_size = data_entry.header.record_size * configs_.pmem_block_size;
+    auto record_size = data_entry.header.record_size;
+    if (record_size > configs_.pmem_segment_blocks * configs_.pmem_block_size) {
```

Comment:
Is this for a merged reused space?
please add comments.

Reply:
done